ginga/canvas/types/astro.py

```diff
@@ -666,7 +679,238 @@ def move_to(self, xdst, ydst):
         self.set_data_points([(xdst, ydst)])
 
 
+class WCSAxes(CompoundObject):
+    """
+    Special compound object to draw WCS axes.
+    """
+    @classmethod
+    def get_params_metadata(cls):
+        return [
+            Param(name='linewidth', type=int, default=1,
+                  min=1, max=20, widget='spinbutton', incr=1,
+                  description="Width of outline"),
+            Param(name='linestyle', type=str, default='dash',
+                  valid=['solid', 'dash'],
+                  description="Style of outline (default dash)"),
+            Param(name='color',
+                  valid=colors_plus_none, type=_color, default='yellow',
+                  description="Color of grid and text"),
+            Param(name='alpha', type=float, default=1.0,
+                  min=0.0, max=1.0, widget='spinfloat', incr=0.05,
+                  description="Opacity of grid and text"),
+            Param(name='font', type=str, default='Sans Serif',
+                  description="Font family for text"),
+            Param(name='fontsize', type=int, default=8,
+                  min=8, max=72,
+                  description="Font size of text (default: 8)"),
+            ]
+
+    def __init__(self, color='cyan',
+                 linewidth=1, linestyle='dash', alpha=1.0,
+                 font='Sans Serif', fontsize=8,
+                 **kwdargs):
+
+        # these could become supplied optional parameters, if desired
+        self.show_label = True
+        self.num_ra = 10
+        self.num_dec = 10
+        self._pix_res = 10
+        self.txt_off = 4
+        self.ra_angle = None
+        self.dec_angle = None
+        # for keeping track of changes to image and orientation
+        self._cur_rot = None
+        self._cur_swap = None
+        self._cur_image = None
+
+        CompoundObject.__init__(self,
+                                color=color, alpha=alpha,
+                                linewidth=linewidth, linestyle=linestyle,
+                                font=font, fontsize=fontsize, **kwdargs)
+
+        self.editable = False
+        self.pickable = False
+        self.opaque = True
+        self.kind = 'wcsaxes'
+
+    def _calc_axes(self, viewer, image, rot_deg, swapxy):
+        self._cur_image = image
+        self._cur_rot = rot_deg
+        self._cur_swap = swapxy
+
+        if not isinstance(image, AstroImage) or not image.has_valid_wcs():
+            return []
+
+        # Approximate bounding box in RA/DEC space
+        xmax = image.width - 1
+        ymax = image.height - 1
+        try:
+            radec = image.wcs.pointstocoords(
+                [[0, 0], [0, ymax], [xmax, 0], [xmax, ymax]],
+                naxispath=image.naxispath)
+        except Exception:
+            return []
+        ra_min, dec_min = radec.ra.min().deg, radec.dec.min().deg
+        ra_max, dec_max = radec.ra.max().deg, radec.dec.max().deg
```

Comment:
Note: Such syntax is very specific to Astropy but was necessary to support image in uncommon WCS system like NCP; Otherwise, Ginga complains about negative RA values and the displayed axes don't match the info bar.